core/kernel/dt.c

```diff
@@ -553,23 +554,35 @@ void init_external_dt(unsigned long phys_dt)
 	if (!IS_ENABLED(CFG_EXTERNAL_DT))
 		return;
 
-	if (!phys_dt) {
+	if (IS_ENABLED(CFG_TRANSFER_LIST))
+		fdt = transfer_list_entry_data(transfer_list_find(dt->tl,
```

Comment:
I'd rather add another argument to `init_external_dt()` so it becomes:
``` C
void init_external_dt(void *fdt, unsigned long phys_dt)
```
That way we don't need to involve transfer lists in this function, instead `fdt` is non-null we use it, else we map the `phys_dt` and use that address.

Reply:
I realize that we already set `boot_arg_fdt` as same as `transfer_list_entry_data(transfer_list_find(dt->tl, TL_TAG_FDT))` via function `boot_save_transfer_list()`, I think we can deprecate this few lines and don't need to add a new argument, too.

Reply:
Ah, no, here I still need to pass an argument to indicate if the DTB is already mapped and bypass the re-map. I will add an arg `void *mapped_dt`.

Reply:
Found a way to check the mapping type of the PA to know whether it is mapped or not, now we don't need to add additional argument to `init_external_dt()`